### PR TITLE
prow/plugins/override: normalize user before looking up in OWNERS

### LINKED ISSUE #15043: Override not allowed, even though username is listed in OWNERS
**What happened**:
`@LorbusChris: LorbusChris unauthorized: /override is restricted to Repo administrators, approvers in top level OWNERS file.`

see https://github.com/openshift/machine-config-operator/pull/1220#issuecomment-548383786

**What you expected to happen**:
Prow let's me do the override 

**How to reproduce it (as minimally and precisely as possible)**:
In MCO repo, it's reproducible for my user on all PRs, another example: https://github.com/openshift/machine-config-operator/pull/1229#issuecomment-548364714

**Please provide links to example occurrences, if any**:
https://github.com/openshift/machine-config-operator/pull/1229#issuecomment-548364714
https://github.com/openshift/machine-config-operator/pull/1220#issuecomment-548383786

**Anything else we need to know?**:
Maybe it's casued by the upper case letters in my username?

### PR DESCRIPTION
**Pull request by steveeJ**
- [x] Add a test case to verify the failure
- [x] Fix the failure

Fixes #15043.

### CODE REVIEW

**Comment by alvaroaleman** [comment]
@steveeJ I'll unassign myself for now, can you assign me once this is ready for a review? Thanks!
/uncc

**Comment by steveeJ** [comment]
@alvaroaleman Thanks for approving the tests. The test case I added failed as expected and demonstrates the bug:

```
    --- FAIL: TestHandle/override_with_allow_top_level_owners_works_for_uppercase_user (0.00s)
        override_test.go:740: bad statuses: actual map[string]github.Status{"job":github.Status{State:"failure", TargetURL:"", Description:"failed", Context:"job"}} != expected map[string]github.Status{"job":github.Status{State:"success", TargetURL:"", Description:"Overridden by code_owner", Context:"job"}} 
```

I'll add a commit which fixes this case to this PR shortly.

**Comment by steveeJ** [comment]
This is ready for review.

/assign @alvaroaleman

File: prow/plugins/override/override_test.go [hunk 1]
```diff
@@ -597,6 +597,27 @@ func TestHandle(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:      "override with allow_top_level_owners works for uppercase user",
```
Comment (alvaroaleman) [inline prow/plugins/override/override_test.go:601]: Whats with the other way round, the user in the owners file is not all lowercase?

Comment (steveeJ) [reply prow/plugins/override/override_test.go:601]: The test function currently differs from the tested code in that aspect. The real code effectively runs `github.NormLogin` on all users which are loaded from the _OWNERS_ file. The test doesn't do that. Being aware of that we don't need to test the case you're mentioning, although I'm open to rewriting the test function to be more similar to the tested code.

Comment (alvaroaleman) [reply prow/plugins/override/override_test.go:601]: okay sgtm. Can youy sign the CLA and squash your commits?

Comment (steveeJ) [reply prow/plugins/override/override_test.go:601]: I squashed the first two commits and added a fixup commit on top that makes the changes to the test function as described above. I think it makes sense to keep that commit and squash it as well, WDTY?
Meanwhile I'll read up on the CLA signature.

Comment (alvaroaleman) [reply prow/plugins/override/override_test.go:601]: What is a fixup commit? I don't think there is value in having the testcase declare an uppercase user just so the test definition normalizes it, that is effectively just testcode testing testcode

Comment (steveeJ) [reply prow/plugins/override/override_test.go:601]: > What is a fixup commit?

It's what I call a commit that was created using `git commit --fixup=<some-rev>`.

> I don't think there is value in having the testcase declare an uppercase user just so the test definition normalizes it, that is effectively just testcode testing testcode

I see that point. Unfortunately the real code is not suited to be called from the test function and I'm currently not motivated to refactor that. I'll drop the fixup commit then.

**Pull request merged**